### PR TITLE
Adds question answering task to Extract information NLP docs

### PR DESCRIPTION
**Pull request by szabosteve**
## Overview

This PR adds:
* a new section to the Extract information page called `Question answering`,
* a short description of the task with an example,
* the tested models to the list of supported third-party models.

Related to https://github.com/elastic/elasticsearch/pull/85958 and https://github.com/elastic/ml-team/issues/794.

### Preview

* [Question answering](https://stack-docs_2138.docs-preview.app.elstc.co/guide/en/machine-learning/master/ml-nlp-extract-info.html#ml-nlp-question-answering)
* [Model reference](https://stack-docs_2138.docs-preview.app.elstc.co/guide/en/machine-learning/master/ml-nlp-model-ref.html#ml-nlp-model-ref-question-answering)

### CODE REVIEW

File: docs/en/stack/ml/nlp/ml-nlp-extract-info.asciidoc [hunk 1]
```diff
@@ -100,4 +101,40 @@ The task returns the following result:
...
+...
+{
+    "text_field": "The Amazon rainforest (Portuguese: Floresta Amazônica or Amazônia; Spanish: Selva Amazónica, Amazonía or usually Amazonia; French: Forêt amazonienne; Dutch: Amazoneregenwoud), also known in English as Amazonia or the Amazon Jungle, is a moist broadleaf forest that covers most of the Amazon basin of South America. This basin encompasses 7,000,000 square kilometres (2,700,000 sq mi), of which 5,500,000 square kilometres (2,100,000 sq mi) are covered by the rainforest. This region includes territory belonging to nine nations. The majority of the forest is contained within Brazil, with 60% of the rainforest, followed by Peru with 13%, Colombia with 10%, and with minor amounts in Venezuela, Ecuador, Bolivia, Guyana, Suriname and French Guiana. States or departments in four nations contain "Amazonas" in their names. The Amazon represents over half of the planet's remaining rainforests, and comprises the largest and most biodiverse tract of tropical rainforest in the world, with an estimated 390 billion individual trees divided into 16,000 species.",
+    "question": "Which name is also used to describe the Amazon rainforest in English?"
+}
```
Comment (benwtrent) [inline docs/en/stack/ml/nlp/ml-nlp-extract-info.asciidoc:125]: Should this look like the `_infer` API? If so, we need to make some changes...

Comment (szabosteve) [reply docs/en/stack/ml/nlp/ml-nlp-extract-info.asciidoc:125]: @benwtrent 
On this page, we use pseudo-examples to make the logic of the tasks more understandable but without the low-level details of the API calls. This is why these examples are marked as `//NOTCONSOLE`, so the rendered page won't urge users to copy/paste the examples.

Comment (benwtrent) [reply docs/en/stack/ml/nlp/ml-nlp-extract-info.asciidoc:125]: ok. 

```
"docs": [{"text_field": <long_text>}],
"inference_config": {"question_answering": {"question": <question>}}
```
Is the correct syntax for what its worth

**Review by benwtrent — APPROVED** [review]

File: docs/en/stack/ml/nlp/ml-nlp-extract-info.asciidoc [hunk 2]
```diff
@@ -100,4 +101,41 @@ The task returns the following result:
...
+[discrete]
+[[ml-nlp-question-answering]]
+== Question answering
+
+The question answering (or extractive question answering) task makes it possible 
+to get answers to certain questions by extracting information from the provided 
+text.
+
+The model tokenizes the string of – usually long – unstructured text, then it 
+attempts to pull the answer from the text to the provided question. The logic is
```
Comment (lcawl) [inline docs/en/stack/ml/nlp/ml-nlp-extract-info.asciidoc:116]: Minor, but this sentence tripped my brain a bit. Perhaps clearer like this?
```suggestion
attempts to pull an answer for your question from the text. The logic is 
```

**Review by lcawl — APPROVED** [review]
Minor suggestion, otherwise LGTM

**Pull request merged**